dbt/adapters/base/impl.py

```diff
@@ -1328,14 +1328,16 @@ def calculate_freshness_from_metadata_batch(
         # Track schema, identifiers of sources for lookup from batch query
         schema_identifier_to_source = {
             (
-                source.path.get_lowered_part(ComponentName.Schema),
-                source.path.get_lowered_part(ComponentName.Identifier),
+                source.path.get_lowered_part(ComponentName.Schema),  # type: ignore
```

Comment:
I'm not sure why `mypy` doesn't like this, but it doesn't. It think it's getting a `str` and wants a `ComponentName`, of which this is clearly an instance. Maybe it has something to do with the fact that we rolled our own `StrEnum`.